pkg/controller/core.oam.dev/v1alpha2/application/apply.go

```diff
@@ -287,122 +204,16 @@ func (h *appHandler) statusAggregate(appFile *appfile.Appfile) ([]common.Applica
-	}
-	if len(preRevisionName) != 0 {
-		needNewRevision, err := utils.CompareWithRevision(ctx, h.r, compName, compNameSpace, preRevisionName, &updatedComp.Spec)
-		if err != nil {
-			return "", errors.Wrap(err, fmt.Sprintf("compare with existing controllerRevision %s failed",
-				preRevisionName))
-		}
-		if !needNewRevision {
-			klog.InfoS("No need to wait for a new component revision", "component", klog.KObj(updatedComp),
-				"revision", preRevisionName)
-			return preRevisionName, nil
-		}
-	}
-	klog.InfoS("Wait for a new component revision", "component name", compName,
-		"previous revision", preRevisionName)
-	// get the new component revision that contains the component with retry
-	checkForRevision := func() (bool, error) {
-		if err := h.r.Get(ctx, compKey, &curComp); err != nil {
-			// retry no matter what
-			// nolint:nilerr
-			return false, nil
-		}
-		if curComp.Status.LatestRevision == nil || curComp.Status.LatestRevision.Name == preRevisionName {
-			return false, nil
-		}
-		needNewRevision, err := utils.CompareWithRevision(ctx, h.r, compName,
-			compNameSpace, curComp.Status.LatestRevision.Name, &updatedComp.Spec)
-		if err != nil {
-			// retry no matter what
-			// nolint:nilerr
-			return false, nil
-		}
-		// end the loop if we find the revision
-		if !needNewRevision {
-			curRevisionName = curComp.Status.LatestRevision.Name
-			klog.InfoS("Get a matching component revision", "component name", compName,
-				"current revision", curRevisionName)
-		}
-		return !needNewRevision, nil
-	}
-	if err := wait.ExponentialBackoff(utils.DefaultBackoff, checkForRevision); err != nil {
-		return "", err
-	}
-	return curRevisionName, nil
-}
-
-func (h *appHandler) applyHelmModuleResources(ctx context.Context, comp *v1alpha2.Component, owners []metav1.OwnerReference) error {
-	klog.Info("Process a Helm module component")
-	repo, err := oamutil.RawExtension2Unstructured(&comp.Spec.Helm.Repository)
-	if err != nil {
-		return err
-	}
-	release, err := oamutil.RawExtension2Unstructured(&comp.Spec.Helm.Release)
-	if err != nil {
-		return err
-	}
-
-	release.SetOwnerReferences(owners)
-	repo.SetOwnerReferences(owners)
-
-	if err := h.r.applicator.Apply(ctx, repo); err != nil {
-		return err
-	}
-	klog.InfoS("Apply a HelmRepository", "namespace", repo.GetNamespace(), "name", repo.GetName())
-	if err := h.r.applicator.Apply(ctx, release); err != nil {
-		return err
-	}
-	klog.InfoS("Apply a HelmRelease", "namespace", release.GetNamespace(), "name", release.GetName())
-	return nil
-}
-
-// checkAutoDetect judge whether the workload type of a helm mode component is not clear, an autodetect type workload
-// will be specified by default In this case, the traits attached to the helm mode component will fail to generate, so
-// we only call applyHelmModuleResources to create the helm resource, don't generate other K8s resources.
-func (h *appHandler) checkAutoDetect(component *v1alpha2.Component) bool {
```

Comment:
the function is more clear, we should keep it

Reply:
reverted.